src/ast/semantic_analyser.cpp

```diff
@@ -604,7 +604,7 @@ void SemanticAnalyser::visit(Call &call)
         }
       }
 
-      if (call.vargs->size() == 2)
+      if (is_final_pass() && call.vargs->size() == 2)
```

Comment:
I think you have to check the type before the downcast otherwise you don't know what you're casting
```suggestion
      if (is_final_pass() && check_arg(..) && call.vargs->size() == 2)
```

And then delete the `check_arg()` below

Reply:
I see. I missed the `check_arg()` below. thanks.